File: app/src/main/java/com/gdi/hahi/mvp/model/NeihanModel.java
package com.gdi.hahi.mvp.model;

import android.content.Context;
import android.util.Log;

import com.gdi.hahi.mvp.contract.HahiContract;
import com.gdi.hahi.mvp.model.bean.HahiBean;
import com.gdi.hahi.mvp.model.bean.Neihan;
import com.gdi.hahi.network.Api;
import com.gdi.hahi.network.RetrofitClient;

import java.util.ArrayList;
import java.util.List;

import io.reactivex.Observable;

/**
 * Created by gandi on 2017/8/24 0024.
 */

public class NeihanModel implements HahiContract.Model<Neihan, List<HahiBean>> {

    private static final String TAG = "NeihanModel";
    String content_type;
    String device_id;

    public NeihanModel(String content_type, String device_id) {
        this.content_type = content_type;
        this.device_id = device_id;
    }

    @Override
    public List<HahiBean> convertToDoMain(Neihan data) {
        List<HahiBean> hahiList = new ArrayList<>();
        if (data.getData()==null){
            return hahiList;
        }
        List<Neihan.DataBeanX.DataBean> data1 = data.getData().getData();
        for (Neihan.DataBeanX.DataBean dataBean : data1) {
            HahiBean hahiBean = new HahiBean();
            hahiBean.setUser_name(dataBean.getGroup().getUser().getName());
            hahiBean.setAvatar_url(dataBean.getGroup().getUser().getAvatar_url());
            hahiBean.setContext(dataBean.getGroup().getContent());
            int meida_type = dataBean.getGroup().getMedia_type();
            switch (meida_type) {
                case 0:
                    break;
                case 1:
                    hahiBean.setMeida_url(dataBean.getGroup().getLarge_image().getUrl_list().get(0).getUrl());
                    break;
                case 2:
                    hahiBean.setMeida_url(dataBean.getGroup().getLarge_image().getUrl_list().get(0).getUrl());
                    hahiBean.setIs_gif(dataBean.getGroup().getIs_gif());
                    hahiBean.setWidth(dataBean.getGroup().getLarge_image().getWidth());
                    hahiBean.setHight(dataBean.getGroup().getLarge_image().getHeight());
                    break;
                case 3:
                    hahiBean.setMeida_url(dataBean.getGroup().getMp4_url());
                    hahiBean.setIs_video(dataBean.getGroup().getIs_video());
                    hahiBean.setWidth(dataBean.getGroup().get_$360p_video().getWidth());
                    hahiBean.setHight(dataBean.getGroup().get_$360p_video().getHeight());
                    hahiBean.setDuration((int) dataBean.getGroup().getDuration());
                    hahiBean.setThumbImage(dataBean.getGroup().getLarge_cover().getUrl_list().get(0).getUrl());
                    hahiBean.setPlay_count(dataBean.getGroup().getPlay_count());
                    break;
            }
            hahiBean.setMedia_type(meida_type);
            Log.i(TAG, "convertToDoMain: " + hahiBean.toString());
            hahiList.add(hahiBean);
        }

        return hahiList;
    }

    @Override
    public Observable<Neihan> loadData(Context context,long lastTime) {
        RetrofitClient retrofitClient = RetrofitClient.getInstance(context, Api.BaseUrl);
        Observable<Neihan> observable = retrofitClient.create(Api.class).getNeiHanData(content_type,device_id,lastTime,System.currentTimeMillis());
        return observable;
    }
}

File: app/src/main/java/com/gdi/hahi/mvp/model/NeihanModel.kt
package com.gdi.hahi.mvp.model

import android.content.Context

import com.gdi.hahi.mvp.contract.HahiContract
import com.gdi.hahi.mvp.model.bean.DataN
import com.gdi.hahi.mvp.model.bean.NeiHanData
import com.gdi.hahi.mvp.model.bean.NeiHanBean
import com.gdi.hahi.api.Api
import com.gdi.lazylibrary.network.RetrofitClient

import io.reactivex.Observable

/**
 * Created by gandi on 2017/8/24 0024.
 */

class NeihanModel(internal var content_type: String, internal var device_id: String) : HahiContract.Model<NeiHanBean, List<NeiHanData>> {

    override fun convertToDoMain(neiHanBean: NeiHanBean): List<NeiHanData>? {
//        val hahiList = ArrayList<NeiHanData>()
        if (neiHanBean.data == null) {
            return null
        }
        return neiHanBean.data.data.map { convertNeiHanDataToDoMain(it) }
    }

    fun convertNeiHanDataToDoMain(dataN: DataN): NeiHanData {
        var group = dataN.group
        var is_gif: Int = 0
        var is_video: Int = 0
        var meida_url: String=""
        var width: Int = 0
        var hight: Int = 0
        var duration: Int = 0
        var play_count: Int = 0
        var thumbImage: String = ""
        
        with(group){
            when (media_type) {
                1 -> {
                    meida_url = large_image.url_list[0].url
                }
                2 -> {
                    meida_url = large_image.url_list[0].url
                    is_gif = is_gif
                    width = large_image.width
                    hight = large_image.height
                }
                3 -> {
                    meida_url = mp4_url
                    is_video = is_video
                    width = video360p.width
                    hight = video360p.height
                    duration = duration.toInt()
                    thumbImage = large_cover.url_list[0].url
                    play_count = play_count
                }
            }
            return NeiHanData(user.name, user.avatar_url, content, media_type, is_gif, is_video, meida_url!!, width, hight, duration, play_count, thumbImage!!)
        }
    }




    override fun loadData(context: Context, lastTime: Long): Observable<NeiHanBean> {
        val retrofitClient = RetrofitClient.getInstance(context).baseUrl(Api.BaseUrl)
        val observable = retrofitClient.create<Api>(Api::class.java).getNeiHanData(content_type, device_id, lastTime, System.currentTimeMillis())
        return observable
    }

    companion object {

        private val TAG = "NeihanModel"
    }
}



